address-space-controller/src/main/java/io/enmasse/controller/EndpointController.java

```diff
@@ -158,54 +159,66 @@ private Route ensureRouteExists(AddressSpace addressSpace, EndpointSpec endpoint
                 .editOrNewMetadata()
                 .withName(routeName)
                 .withNamespace(namespace)
+                .addToAnnotations(exposeSpec.getAnnotations() != null ? exposeSpec.getAnnotations() : Collections.emptyMap())
                 .addToAnnotations(AnnotationKeys.ADDRESS_SPACE, addressSpace.getName())
                 .addToAnnotations(AnnotationKeys.SERVICE_NAME, serviceName)
                 .addToLabels(LabelKeys.INFRA_TYPE, addressSpace.getType())
                 .addToLabels(LabelKeys.INFRA_UUID, infraUuid)
                 .endMetadata()
                 .editOrNewSpec()
-                .withHost(endpointSpec.getHost().orElse(""))
+                .withHost(endpointSpec.getExposeSpec().flatMap(ExposeSpec::getRouteHost).orElse(""))
                 .withNewTo()
                 .withName(serviceName)
                 .withKind("Service")
                 .endTo()
                 .withNewPort()
                 .editOrNewTargetPort()
-                .withStrVal(endpointSpec.getServicePort())
+                .withStrVal(exposeSpec.getRouteServicePort())
                 .endTargetPort()
                 .endPort()
                 .endSpec();
 
+
         if (endpointSpec.getCertSpec().isPresent()) {
+            ExposeSpec.TlsTermination tlsTermination = exposeSpec.getRouteTlsTermination();
             CertSpec certSpec = endpointSpec.getCertSpec().get();
-            if ("https".equals(endpointSpec.getServicePort()) && "selfsigned".equals(certSpec.getProvider())) {
-                String caSecretName = KubeUtil.getAddressSpaceExternalCaSecretName(addressSpace);
-                Secret secret = client.secrets().inNamespace(namespace).withName(caSecretName).get();
-                if (secret != null) {
-                    String consoleCa = new String(Base64.getDecoder().decode(secret.getData().get("tls.crt")), StandardCharsets.UTF_8);
+
+            if (tlsTermination.equals(ExposeSpec.TlsTermination.passthrough)) {
```

Comment:
Can tlsTermination be legally null? If yes, NPE is possible here.  If no then I think we should be failing early by validation the object when built by the builder.

Reply:
I will validate it in the builder given the expose type.

Reply:
Added validation in builder.